presto-main/src/test/java/com/facebook/presto/execution/TestUseTask.java

```diff
@@ -0,0 +1,168 @@
+        metadata = createTestMetadataManager(transactionManager);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        executor.shutdownNow();
+    }
+
+    @Test
+    public void testUse()
+    {
+        Use use = new Use(Optional.of(identifier("test_catalog")), identifier("test_schema"));
+        String sqlString = "USE test_catalog.test_schema";
+        executeUse(use, sqlString, TEST_SESSION);
+    }
+
+    @Test(
+            expectedExceptions = SemanticException.class,
+            expectedExceptionsMessageRegExp = "Catalog must be specified when session catalog is not set")
+    public void testUseNoCatalog()
+    {
+        Use use = new Use(Optional.empty(), identifier("test_schema"));
+        String sqlString = "USE test_schema";
+        Session session = testSessionBuilder()
+                .setCatalog(null)
+                .setSchema(null)
+                .build();
+        executeUse(use, sqlString, session);
+    }
+
+    @Test(
+            expectedExceptions = SemanticException.class,
+            expectedExceptionsMessageRegExp = "Catalog does not exist: invalid_catalog")
+    public void testUseInvalidCatalog()
+    {
+        Use use = new Use(Optional.of(identifier("invalid_catalog")), identifier("test_schema"));
+        String sqlString = "USE invalid_catalog.test_schema";
+        executeUse(use, sqlString, TEST_SESSION);
+    }
+
+    @Test(
+            expectedExceptions = SemanticException.class,
+            expectedExceptionsMessageRegExp = "Schema does not exist: test_catalog.invalid_schema")
+    public void testUseInvalidSchema()
+    {
+        Use use = new Use(Optional.of(identifier("test_catalog")), identifier("invalid_schema"));
+        String sqlString = "USE test_catalog.invalid_schema";
+        Session session = testSessionBuilder()
+                .setSchema("invalid_schema")
+                .build();
+        executeUse(use, sqlString, session);
+    }
+
+    @Test(
+            expectedExceptions = AccessDeniedException.class,
+            expectedExceptionsMessageRegExp = "Access Denied: Cannot access catalog test_catalog")
+    public void testUseAccessDenied()
+    {
+        Use use = new Use(Optional.of(identifier("test_catalog")), identifier("test_schema"));
+        String sqlString = "USE test_catalog.test_schema";
+        Session session = testSessionBuilder()
+                .setIdentity(new Identity("user", Optional.empty()))
+                .build();
+        AccessControl accessControl = new DenyAllAccessControl();
+        executeUse(use, sqlString, session, accessControl);
+    }
+
+    private void executeUse(Use use, String sqlString, Session session)
+    {
+        executeUse(use, sqlString, session, new AllowAllAccessControl());
+    }
+
+    private void executeUse(Use use, String sqlString, Session session, AccessControl accessControl)
+    {
+        catalogManager = new CatalogManager();
```

Comment:
Since we have already created these in the beforeClass method, do we need to create them again here?

Reply:
since we gave that only in beforeClass we need to give that in execute () otherwise any changes made by one test (e.g., catalog registrations, state updates) can interfere with subsequent tests.